cpp/src/strings/regex/regex_program_impl.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "regcomp.h"
+#include "regex.cuh"
+
+#include <cudf/strings/regex/regex_program.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+namespace strings {
+
+/**
+ * @brief Implementation object for regex_program
+ *
+ * It encapsulates internal reprog object used for building its device equivalent
+ */
+struct regex_program::regex_program_impl {
+  detail::reprog prog;
+
+  // TODO: There will be other options added here in the future to handle issues
+  // 10852 and possibly others like 11979
+
```

Comment:
If this is a note to reviewers, thanks! Should we remove it before merging?

Reply:
No, should we?

Reply:
No, if you think it's helpful to leave here that's fine with me.